ait/core/api.py

```diff
@@ -554,36 +555,95 @@ def start (self):
         super(UdpTelemetryServer, self).start()
 
 
+class TlmMonitor(gevent.Greenlet):
+    def __init__(self, pktbufs, defns):
+        gevent.Greenlet.__init__(self)
+        self._pktbufs = pktbufs
+        self._defns = defns
 
-class Instrument (object):
-    def __init__(self,
-                 cmdport=ait.config.get('command.port', ait.DEFAULT_CMD_PORT),
-                 tlmport=3076, defn=None):
-        if defn is None:
-            tlmdict = tlm.getDefaultDict()
-            names   = sorted( tlmdict.keys() )
+        self._cntxt = zmq.Context()
+        sub_url = ait.SERVER_DEFAULT_XPUB_URL.replace('*', 'localhost')
 
-            if len(names) == 0:
-                msg = 'No packets defined in default TLM dictionary.'
-                raise TypeError(msg)
+        self._sub = self._cntxt.socket(zmq.SUB)
+        self._sub.connect(sub_url)
+        self._sub.setsockopt_string(zmq.SUBSCRIBE, ait.DEFAULT_TLM_TOPIC)
+
+    def _run(self):
+        try:
+            while True:
+                gevent.sleep(0)
+                raise Exception()
```

Comment:
@MJJoyce Is this for active dev-debugging?

Reply:
Bleh ya, looks like I failed to save properly before pushing the branch up. That's what I get for trying to get up before I ran out the door. I'll fix it up tomorrow along with some other changes to tests and whatnot. Good catch!